File: library_zap/src/main/java/com/dolin/zap/Zap.java
package com.dolin.zap;

import android.app.Application;

import com.dolin.zap.entity.Config;
import com.dolin.zap.impl.ZapPrint;

/**
 * @author #Suyghur.
 * Created on 2021/05/19
 */
public class Zap {

    private static boolean hasInitialized = false;

    public static void initialize(Application application, Config config) {
        ZapPrint.getInstance().initialize(application, config);
        hasInitialized = true;
    }

    public static void recycle() {
        if (hasInitialized) {
            ZapPrint.getInstance().recycle();
        }
    }

    public static void d(Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().d(obj);
        }
    }

    public static void d(String tag, Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().d(tag, obj);
        }
    }

    public static void i(Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().i(obj);
        }
    }

    public static void i(String tag, Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().i(tag, obj);
        }
    }

    public static void w(Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().w(obj);
        }
    }

    public static void w(String tag, Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().w(tag, obj);
        }
    }

    public static void e(Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().e(obj);
        }
    }

    public static void e(String tag, Object obj) {
        if (hasInitialized) {
            ZapPrint.getInstance().e(tag, obj);
        }
    }

    public static String getVersion() {
        return Version.VERSION;
    }

    public static String getFullVersion() {
        return Version.FULL_VERSION;
    }
}

File: library_zap/src/main/java/com/dolin/zap/Zap.kt
package com.dolin.zap

import android.app.Application
import com.dolin.zap.entity.Config
import com.dolin.zap.impl.ZapPrint

/**
 * @author #Suyghur.
 * Created on 4/7/21
 */
object Zap {

    private lateinit var sZapPrint: ZapPrint
    private var hasInitialized = false

    @JvmStatic
    fun initialize(application: Application, config: Config) {
        sZapPrint = ZapPrint.instance()
        sZapPrint.initialize(application, config)
        hasInitialized = true
    }

    @JvmStatic
    fun recycle() {
        if (hasInitialized)
            sZapPrint.recycle()
    }

    @JvmStatic
    fun d(any: Any?) {
        if (hasInitialized)
            sZapPrint.d(any)
    }

    @JvmStatic
    fun d(tag: String, any: Any?) {
        if (hasInitialized)
            sZapPrint.d(tag, any)
    }

    @JvmStatic
    fun i(any: Any?) {
        if (hasInitialized)
            sZapPrint.i(any)
    }

    @JvmStatic
    fun i(tag: String, any: Any?) {
        if (hasInitialized)
            sZapPrint.i(tag, any)
    }

    @JvmStatic
    fun w(any: Any?) {
        if (hasInitialized)
            sZapPrint.w(any)
    }

    @JvmStatic
    fun w(tag: String, any: Any?) {
        if (hasInitialized)
            sZapPrint.w(tag, any)
    }

    @JvmStatic
    fun e(any: Any?) {
        if (hasInitialized)
            sZapPrint.e(any)
    }

    @JvmStatic
    fun e(tag: String, any: Any?) {
        if (hasInitialized)
            sZapPrint.e(tag, any)
    }

    @JvmStatic
    fun getVersion(): String {
        return Version.VERSION
    }

    @JvmStatic
    fun getFullVersion(): String {
        return Version.FULL_VERSION
    }
}